src/objective/regression_objective.hpp

```diff
@@ -343,6 +360,25 @@ class RegressionPoissonLoss: public ObjectiveFunction {
 
   bool BoostFromAverage() const override { return true; }
 
+  bool GetCustomAverage(double *initscore) const override {
+    if (initscore == nullptr) return false;
+    double sumw = 0.0f;
+    double sumy = 0.0f;
+    if (weights_ == nullptr) {
+      for (int i = 0; i < num_data_; i++) sumy += label_[i];
```

Comment:
I think it's better for us to follow the cpp coding styles in other soruce codes such like https://github.com/Microsoft/LightGBM/blob/c62dcf7319300a82d1ca8d1df720e359fdb658ff/src/objective/xentropy_objective.hpp#L122-L124

i.e.

```cpp
      for (int i = 0; i < num_data_; i++) {
          sumy += label_[i];
      }
```

However, if @guolinke doesn't care this, I'm ok.